glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -262,30 +258,21 @@ open class GleanInternalAPI internal constructor () {
 
             @Suppress("EXPERIMENTAL_API_USAGE")
             Dispatchers.API.launch {
-                // glean_set_upload_enabled might delete all of the queued pings. We
-                // therefore need to obtain the lock from the PingUploader so that
-                // iterating over and deleting the pings doesn't happen at the same time.
-                synchronized(PingUploadWorker.pingQueueLock) {
-                    LibGleanFFI.INSTANCE.glean_set_upload_enabled(enabled.toByte())
+                LibGleanFFI.INSTANCE.glean_set_upload_enabled(enabled.toByte())
```

Comment:
Please comment on why it's safe to remove synchronized here.

Reply:
Wouldn't a comment like that be a little confusing? The lock doesn't even exist anymore, all that synchronization is done on the rust side (https://github.com/mozilla/glean/blob/m16/ping-uploading/glean-core/src/lib.rs#L329).

Reply:
Sorry, I hastily commented :) I meant "can you explain why it's safe to remove synchronized"? Not add a comment :D

Reply:
It's safe because rust ensures it is synchronized :)